TidepoolKit/Model/Datum/DeviceEventDatum/TTimeChangeDeviceEventDatum.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  TTimeChangeDeviceEventDatum.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 11/1/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public class TTimeChangeDeviceEventDatum: TDeviceEventDatum, Decodable {
+    public enum Method: String, Codable {
+        case automatic
+        case manual
+    }
+
+    public var from: Info?
+    public var to: Info?
+    public var method: Method?
+    public var change: Change?  // DEPRECATED
+
+    public init(time: Date, from: Info, to: Info, method: Method? = nil) {
+        self.from = from
+        self.to = to
+        self.method = method
+        super.init(.timeChange, time: time)
+    }
+
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.from = try container.decodeIfPresent(Info.self, forKey: .from)
+        self.to = try container.decodeIfPresent(Info.self, forKey: .to)
+        self.method = try container.decodeIfPresent(Method.self, forKey: .method)
+        self.change = try container.decodeIfPresent(Change.self, forKey: .change)
+        try super.init(.timeChange, from: decoder)
+    }
+
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeIfPresent(from, forKey: .from)
+        try container.encodeIfPresent(to, forKey: .to)
+        try container.encodeIfPresent(method, forKey: .method)
+        try container.encodeIfPresent(change, forKey: .change)
+        try super.encode(to: encoder)
+    }
+
+    public struct Info: Codable, Equatable {
+        public var time: String?
+        public var timeZoneName: String?
+
+        public init(time: String? = nil, timeZoneName: String? = nil) {
+            self.time = time
+            self.timeZoneName = timeZoneName
```

Comment:
I'm not aware of any pumps that actually have timezone capability; they run on a 24 hour clock that is only changed when the user or controller changes the time. The time on events reported from the pump can be converted to a universal time by using a fixed offset.  Using an actual timezone for this conversion will lead to errors, when the timezone changes its offset, but the pump doesn't.  Is the plan here to store fixed offset timezone names, like "UTC−06:00"? If so, we should document which format is expected here, since there are two unfortunately counterintuitive conventions: "ETC/GMT+6" refers to the same fixed offset timezone as "UTC−06:00". Notice the sign is different.

Reply:
Thanks for explaining how this field is used. That all makes sense, I think.

In addition to device time change, and user timezone change, there is a third item that is probably worth discussing a bit here:

Loop tracks a utc offset that is used for inferring schedule inflection points, and inferring pump event times.  This offset can change when the user syncs time in Loop, if the current user's timezone offset is different than the pump's utc offset. This changes independently of the user's timezone changing. All time inferences using Loop's schedules need to use this pump utc offset. I would think we *do* have some need to make inferences using Loop's configured schedules on the backend, so we need a way to track them. I assume those will be uploaded with schedules (which should trigger a change record when the utc offset is changed).

If that assumption is correct, then I don't have any more questions.  Thanks again, Darin.

Reply:
If I understand correctly, I think I've already accounted the third item. 😄 

In the Tidepool data model `pumpSettings` type (which is populated from `StoredSettings` in Tidepool Loop), there is a new-for-Tidepool-Loop field called `scheduleTimeZoneOffset` that accounts for the pump UTC offset. 

(It used to be split into 4 fields in a previous PR - one for each schedule: basal rates, bg targets, carb ratios, and ISFs - but I have since collapsed that into one common field for all schedules and that collapsed changed will be in a future PR).

Any time the Loop settings are changed a new `StoredSettings` would be saved off and a new `pumpSettings` (among other things) would be uploaded to the backend.

Does that make sense and does that cover your concern?

Reply:
Yes, sounds like what you describe matches my assumption. Thanks!

Reply:
I just want to add that I +1 the notion of putting some documentation comment in here that `timeZoneName` expects an "official" time zone name, please.  IANA does define them (https://www.iana.org/time-zones).  Even special cases like Arizona (which does not observe DST changes) and the Navajo nation (which does, despite being in Arizona) are accounted for there.  (also see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)